src/main/scala/game/PlayerProcess.scala

```diff
@@ -61,5 +85,54 @@ object PlayerProcess {
           _            <- EitherT.fromEither[F](teamsPlayers.map(validatePlayers).sequence)
         } yield ()
       }
+
+      override def finishTurn(playerId: PlayerId): ErrorOrT[F, Unit] = {
+        for {
+          player <- getPlayer(playerId)
+          team   <- teamProcess.getTeamById(player.teamId)
+          game   <- gameProcess.getGameById(player.gameId)
+          _      <- EitherT.fromEither[F](_validateFinishTurnRights(player, team, game))
+          _      <- gameProcess.nextTurn(player.gameId)
+        } yield ()
+      }
+
+      private def _validateFinishTurnRights(player: Player, team: Team, game: Game): Either[GameError, Unit] = {
+        if (game.turn.rightForChangingTurn.equals(player.role) && game.turn.teamColor.equals(team.teamColor)) ().asRight
+        else CannotChangeTurnError.asLeft
+      }
+
+      override def playCard(playerId: PlayerId, cardId: CardId): ErrorOrT[F, Unit] = {
+        for {
+          player     <- getPlayer(playerId)
+          playerTeam <- teamProcess.getTeamById(player.teamId)
+          game       <- gameProcess.getGameById(player.gameId)
+          card       <- cardProcess.getCardById(cardId)
+          _          <- EitherT.liftF(cardProcess.revealCard(card))
+          _          <- _processPlayCardResult(game, playerTeam, card)
+        } yield ()
+      }
+
+      private def _processPlayCardResult(game: Game, team: Team, card: Card): ErrorOrT[F, Unit] = {
+        card.cardRole match {
+          case CardRole.Agent(teamColor) =>
+            for {
+              enemyTeam     <- teamProcess.getEnemyTeam(game.id, team.id)
+              teamPair      <- EitherT.liftF(
```

Comment:
Wrapping into effect (`EitherT.liftF` + `.pure[F]`) is redundant
You can use `=` in for comprehension if your computation does not produce one (effect)